### ISSUE TITLE
Evaluate JobIntentService SecurityException fix

### ISSUE THREAD

**Issue by UweTrottmann** [UweTrottmann/SeriesGuide#575]
https://github.com/UweTrottmann/SeriesGuide/pull/574

Wait until next release, then evaluate crash logs.

If successful suggest fix upstream (support library).

Note: some of these crashes might come from extensions as they use `JobIntentService` as well.

**Comment by UweTrottmann** [comment]
Well, that didn't work:
```
Caused by java.lang.SecurityException: Caller no longer running, last stopped +10s498ms because: timed out while starting
       at android.os.Parcel.readException(Parcel.java:1943)
       at android.os.Parcel.readException(Parcel.java:1889)
       at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:191)
       at android.app.job.JobParameters.dequeueWork(JobParameters.java:196)
       at android.support.v4.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
       at android.support.v4.app.JobIntentService.dequeueWork(JobIntentService.java:640)
       at android.support.v4.app.SafeJobIntentService$SafeCommandProcessor.doInBackground(SafeJobIntentService.java:78)
       at android.support.v4.app.SafeJobIntentService$SafeCommandProcessor.doInBackground(SafeJobIntentService.java:68)
```

Guess we have try+catch until maybe https://issuetracker.google.com/issues/63622293 is fixed.

**Comment by UweTrottmann** [comment]
Actually the current implementation (28.0.0) did not benefit from the previous (#574) change.

So: https://github.com/UweTrottmann/SeriesGuide/pull/587

**Comment by UweTrottmann** [comment]
Nope, still seeing crashes. Though this might also just be extensions that are called by the app that do not use the apps implementation...

Leaving this for now as is.

**Comment by UweTrottmann** [comment]
Look into using WorkManager to replace JobIntentService. But that will definitely break backwards-compatibility. So at least for extensions shipped with the app?

**Comment by UweTrottmann** [comment]
Modified the extension action receiver to use a coroutine instead of a JobIntentService.

**Comment by UweTrottmann** [comment]
Looking at crashes they seem to happen on two occasions:
- directly after app launch
  - Initialize ExtensionManager only when first used. https://github.com/UweTrottmann/SeriesGuide/commit/a4ae6688f75a7bd7438943a9df85abb12b4ace9f
- after `ACTION_PACKAGE_CHANGED` on our own package.
  - Apparently WorkManager enables a RescheduleReceiver it registers, triggering the broadcast. Update the ExtensionPackageChangeReceiver to ignore changes of the app package. https://github.com/UweTrottmann/SeriesGuide/commit/f3bb005cbb575318a09be3bfeb63010965bca2b6

**Comment by UweTrottmann** [comment]
Number of crashes reduced significantly (e.g. 19 vs 2 for v49 vs v50 over 90 days).

Side note: also dropped WorkManager in favor of coroutines as it had issues on some devices:

Galaxy Tab A (2016) (gtaxlwifi)
`java.lang.NoSuchMethodError` when trying to call `android.app.ActivityManager#isLowRamDevice`. However this method should be available since Android 4.4.
https://discuss.seriesgui.de/t/crashes-on-open/339/5?u=uwe
